### ISSUE TITLE
Test suite does not compile extracted code

### ISSUE THREAD

**Issue by Zimmi48** [Zimmi48/bugzilla-test#5668]
Note: the issue was created automatically with bugzilla2github tool

# Bugzilla Bug ID: 5668
Date: 2017-07-26 15:38:27 +0200
From: @_tchajed
To:   
CC:   coq-bugs-redist@lists.gforge.inria.fr

Last updated: 2017-07-27 17:51:14 +0200

## Bugzilla Comment ID: 17175
Date: 2017-07-26 15:38:27 +0200
From: @_tchajed

The test suite does not compile any extracted code in OCaml or Haskell, which 
is concerning for regressions. For example, PR 918 
(https://github.com/coq/coq/pull/918/files) fixes bugs 4844 and 4824 but the 
test doesn't really confirm the issue is fixed since the result code needs to 
be compiled with GHC. Pierre mentioned this shortcoming in the pull request; 
I'm just creating an issue to track this.

Some other random issues related to extracted code that doesn't compile include 
5397 and 5274.

## Bugzilla Comment ID: 17188
Date: 2017-07-27 17:51:14 +0200
From: @_letouzey

See my comment in PR 918 : https://github.com/coq/coq/pull/918#issuecomment-318401414

Short version:
 - for OCaml there is now the Extraction TestCompile command 
 - for Haskell it should *not* be part of the main test-suite of Coq, but apart from that it could be cool to have this kind of compilation tests for Haskell extracted code. I probably won't work on that anytime soon anyway.
